### PR TITLE
Fix tests around the dimensions test data.

### PR DESCRIPTION
**Pull request by forus**

### CODE REVIEW

File: transmart-core-db-tests/src/main/groovy/org/transmartproject/db/StudyTestData.groovy [hunk 1]
```diff
@@ -21,37 +19,17 @@ class StudyTestData {
...
-                assert false, "Dimensions iterable passed to createStudy contains object that is not a String or a " +
-                        "DimensionDescription: $it"
-            }
-        }
-        study
+    static Study createStudy(String name, List<String> dimensions = [], boolean isPublic = false) {
+        new Study(
+            studyId: name,
+            secureObjectToken: isPublic ? Study.PUBLIC : "EXP:${name}",
+            dimensionDescriptions: DimensionDescription.findAllByNameInList(dimensions)
```
Comment (JanKanis) [inline transmart-core-db-tests/src/main/groovy/org/transmartproject/db/StudyTestData.groovy]: I think this needs a check that all requested dimensions are actually found

**Comment by JanKanis** [comment]
Apart from that comment it looks fine

File: transmart-core-db-tests/src/main/groovy/org/transmartproject/db/StudyTestData.groovy (hunk 1 above)
Comment (forus) [inline transmart-core-db-tests/src/main/groovy/org/transmartproject/db/StudyTestData.groovy]: Done

**Comment by forus** [comment]
The code will try to fill the dictionaries on the production run as well.

**Pull request closed without merging**